Example/ViewController.swift

```diff
@@ -200,23 +220,40 @@ class ViewController: UIViewController {
 
     func startBasicNavigation() {
         guard let route = routes?.first else { return }
-
-        let options = NavigationOptions(navigationService: navigationService())
-        let navigationViewController = NavigationViewController(for: route, options: options)
-        navigationViewController.delegate = self
-        navigationViewController.mapView?.delegate = self
         
-        presentAndRemoveMapview(navigationViewController)
+        let service = navigationService(route: route)
+        let navigationViewController = createNavigationViewController(navigationService: service)
+        
+        presentAndRemoveMapview(navigationViewController, completion: beginCarPlayNavigation)
     }
     
     func startNavigation(styles: [Style]) {
         guard let route = routes?.first else { return }
         
-        let options = NavigationOptions(styles: styles, navigationService: navigationService())
+        let options = NavigationOptions(styles: styles, navigationService: navigationService(route: route))
         let navigationViewController = NavigationViewController(for: route, options: options)
         navigationViewController.delegate = self
         
-        presentAndRemoveMapview(navigationViewController)
+        presentAndRemoveMapview(navigationViewController, completion: beginCarPlayNavigation)
+    }
+    
+    func createNavigationViewController(navigationService: NavigationService) -> NavigationViewController {
+        let route = navigationService.route
+        let options = NavigationOptions( navigationService: navigationService)
+        
+        let nav = NavigationViewController(for: route, options: options)
+        nav.delegate = self
+        nav.mapView?.delegate = self
+        return nav
+    }
+    
+    public func beginNavigationWithCarplay(navigationService: NavigationService) {
+        if activeNavigationViewController != nil {
+            activeNavigationViewController?.isUsedInConjunctionWithCarPlayWindow = true
+        } else {
+            let navigationViewController = createNavigationViewController(navigationService: navigationService)
+            presentAndRemoveMapview(navigationViewController, completion: nil)
```

Comment:
@JThramer should we be updating the property `isUsedInConjunctionWithCarPlayWindow` to `true` after we create a new a navigation view controller?

Reply:
Doing it this way makes sure that the StepsViewController is fully presented when the NVC presents, which is less visually jarring IMHO.